### PR TITLE
Type Update

### PR DESCRIPTION
**Pull request by nibhar**
this PR updates the request types used.

- types are separated into public and non-public types with their own files each
- the AccountsRequest class has its own file
- public requests no longer have the `kind` field as it was redundant.
- adds BasicRequest, ParsedBasicRequest, SimpleRequest and ParsedSimpleRequest interfaces and inherits from them

resolves all points of #168

### CODE REVIEW

**Comment by sisou** [comment]
What do you think about renaming the `AccountsRequest` class to `RequestParser`, just like in the keyguard? I think the name AccountsRequest does not tell what the class does.

**Comment by nibhar** [comment]
I wanted to keep the changes to a minimum, but I would agree.

File: src/lib/AccountsRequest.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    ParsedSimpleRequest,
+    ParsedSignTransactionRequest,
+    ParsedCheckoutRequest,
+    ParsedSignMessageRequest,
+    ParsedRenameRequest,
+    ParsedRpcRequest,
+} from './RequestTypes';
+
+export class AccountsRequest {
+    public static parse(request: RpcRequest, state: State, requestType?: RequestType): ParsedRpcRequest | null {
```
Comment (sisou) [inline src/lib/AccountsRequest.ts]: The `requestType` parameter should no longer be optional.

File: src/lib/RpcApi.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ import { BrowserDetection } from '@nimiq/utils';
 import { RootState } from '@/store';
 import { Store } from 'vuex';
 import Router from 'vue-router';
-import { AccountsRequest, RequestType, RpcRequest, RpcResult } from '@/lib/RequestTypes';
+import { RequestType } from './RequestTypes';
+import { AccountsRequest } from './AccountsRequest';
+import { RpcRequest, RpcResult } from './PublicRequestTypes';
```
Comment (sisou) [inline src/lib/RpcApi.ts]: I _really_ dont like that the request types are now split between two files, public and internal. Is that really necessary? What benefits does it bring?

Comment (Bettelstab) [reply src/lib/RpcApi.ts]: It clearly separates the types which are relevant for users of the client from the ones which are just internally used.

**Pull request merged**